yb-voyager/src/srcdb/yugabytedb.go

```diff
@@ -766,14 +766,56 @@ WHERE parent.relname='%s' AND nmsp_parent.nspname = '%s' `, tname, sname)
 	return partitions
 }
 
+// query retrieves all unique columns in the specified tables and schemas, handling both unique constraints and unique indexes, while excluding primary key columns.
 const ybQueryTmplForUniqCols = `
-SELECT tc.table_schema, tc.table_name, kcu.column_name
-FROM information_schema.table_constraints tc
-JOIN information_schema.key_column_usage kcu
-    ON tc.constraint_name = kcu.constraint_name
-	AND tc.table_schema = kcu.table_schema
-    AND tc.table_name = kcu.table_name
-WHERE tc.table_schema = ANY('{%s}') AND tc.table_name = ANY('{%s}') AND tc.constraint_type = 'UNIQUE';
+WITH unique_constraints AS (
+	-- Retrieve columns with unique constraints
+    SELECT
+        tc.table_schema,
+        tc.table_name,
+        kcu.column_name
+    FROM
+        information_schema.table_constraints tc
+    JOIN
+        information_schema.key_column_usage kcu
+        ON tc.constraint_name = kcu.constraint_name
+        AND tc.table_schema = kcu.table_schema
+        AND tc.table_name = kcu.table_name
+    WHERE
+        tc.constraint_type = 'UNIQUE'
+        AND tc.table_schema = ANY('{%s}')
+        AND tc.table_name = ANY('{%s}')
+),
+unique_indexes AS (
+	-- Retrieve columns with unique indexes (excluding primary keys)
+    SELECT
+        n.nspname AS table_schema,
+        t.relname AS table_name,
+        a.attname AS column_name
+    FROM
+        pg_index ix
+	-- Join to get table and schema information from the index
+    JOIN
+        pg_class t ON t.oid = ix.indrelid
+    JOIN
+        pg_namespace n ON n.oid = t.relnamespace
+	-- Join to get column information
+    JOIN
+        pg_attribute a ON a.attrelid = t.oid 
+		AND a.attnum = ANY(ix.indkey)  -- Match indexed columns
+	 -- Left join to ensure we exclude primary keys by checking associated constraints
+    LEFT JOIN
+        pg_constraint c ON ix.indexrelid = c.conindid AND c.contype = 'p'
+    WHERE
+        ix.indisunique = TRUE
+		AND c.contype IS NULL -- Ensure it's not a primary key
```

Comment:
nit: An easier-to-understand way to write this would be to LEFT join based only on `ix.indexrelid = c.conindid` AND fileter out contype='p' in the WHERE

Reply:
yeah what can be done here:
```
-- this ensure all rows of pg_index are there with the contype as 'p' or NULL I
 LEFT JOIN
        pg_constraint c ON ix.indexrelid = c.conindid
        
 -- filter based on contype to exclude PKs
 WHERE c.contype IS NULL; 
```

Reply:
Shouldn't it be 

```
-- this ensure all rows of pg_index are there with the contype as 'p' or NULL I
 LEFT JOIN
        pg_constraint c ON ix.indexrelid = c.conindid
        
 -- filter based on contype to exclude PKs
 WHERE c.contype != 'p'; 
```

Reply:
Maybe that will also filter out the rows where contype is NULL, something we don't want..